tensorboard/plugins/audio/summary_v2.py

```diff
@@ -103,15 +107,31 @@ def lazy_tensor():
             tf.debugging.assert_rank(data, 3)
             tf.debugging.assert_non_negative(max_outputs)
             limited_audio = data[:max_outputs]
+
             encode_fn = functools.partial(
                 audio_ops.encode_wav, sample_rate=sample_rate
             )
-            encoded_audio = tf.map_fn(
-                encode_fn,
-                limited_audio,
-                dtype=tf.string,
-                name="encode_each_audio",
-            )
+            if lengths is not None:
```

Comment:
All else equal, it would be nice to avoid introducing this new control
flow branch. One alternative could be to handle the `lengths is None`
case by synthesizing something like `lengths = int32_max * ones([k])`.
Something like this:

```diff
diff --git a/tensorboard/plugins/audio/summary_v2.py b/tensorboard/plugins/audio/summary_v2.py
index 0efd6e2e4..82cb4ba53 100644
--- a/tensorboard/plugins/audio/summary_v2.py
+++ b/tensorboard/plugins/audio/summary_v2.py
@@ -111,27 +111,22 @@ def audio(
             encode_fn = functools.partial(
                 audio_ops.encode_wav, sample_rate=sample_rate
             )
+            k = tf.shape(limited_audio)[0]
             if lengths is not None:
-                tf.debugging.assert_rank(lengths, 1)
-                limited_lengths = lengths[:max_outputs]
+                limited_lengths = lengths[:k]
+            else:
+                limited_lengths = tf.ones([k], dtype=tf.int64) * (2 ** 63 - 1)
 
-                def encode_with_length(datum_length):
-                    datum, length = datum_length
-                    return encode_fn(datum[:length])
+            def encode_with_length(datum_length):
+                datum, length = datum_length
+                return encode_fn(datum[:length])
 
-                encoded_audio = tf.map_fn(
-                    encode_with_length,
-                    (limited_audio, limited_lengths),
-                    dtype=tf.string,
-                    name="encode_each_audio",
-                )
-            else:
-                encoded_audio = tf.map_fn(
-                    encode_fn,
-                    limited_audio,
-                    dtype=tf.string,
-                    name="encode_each_audio",
-                )
+            encoded_audio = tf.map_fn(
+                encode_with_length,
+                (limited_audio, limited_lengths),
+                dtype=tf.string,
+                name="encode_each_audio",
+            )
             # Workaround for map_fn returning float dtype for an empty elems input.
             encoded_audio = tf.cond(
                 tf.shape(input=encoded_audio)[0] > 0,
```

What do you think?

Reply:
I'm happy to go with that if you prefer -- the length-less codepath produces a smaller and slightly more efficient graph with only a little bit of increased Python complexity so that's why I did it this way.